Cargo.toml

```diff
@@ -51,3 +51,10 @@ hex-literal = "0.4"
 
 [badges]
 travis-ci = { repository = "rusticata/pcap-parser" }
+
+[package.metadata.cargo_check_external_types]
+allowed_external_types = [
+  "nom",
+  "nom::*",
+  "circular::Buffer",
```

Comment:
Just want to call out here: I'm taking the approach of whitelisting everything that is already part of the API. It might make sense to revisit in the future.

Reply:
For the moment, yes. In the future I really need to think about that, but I think this is required here.

The main reason is that a top-level crate can include several parsers, and even combine them (well, that's the point for parser combinators :) ). If crates expose only local types, it makes combination impossible without converting.
For example, the [`IResult`](https://docs.rs/nom/latest/nom/type.IResult.html) is exactly a type that must be shared across parsers (it's not just a `Result`, the `Err` type is specific).